include/sparrow/array_data.hpp

```diff
@@ -197,6 +197,58 @@ namespace sparrow
     template <class L>
     void swap(reference_proxy<L>& lhs, reference_proxy<L>& rhs);
 
+    /*
+     * Layout iterator class
```

Comment:
Could we have a brief documentation of this iterator?
```suggestion
     * Layout iterator class
     *
     * Relies on a layout's couple of value iterator and bitmap iterator to
     * return reference proxies when it is dereferenced.
```